apisix/utils/batch-processor.lua

```diff
@@ -146,8 +150,19 @@ function batch_processor:push(entry)
         return
     end
 
+    if not batch_metrics and prometheus.get_prometheus() and self.name
+       and self.route_id and self.server_addr then
+        batch_metrics = prometheus.get_prometheus():gauge("batch_process_entries",
+                                                          "batch process remaining entries",
+                                                          {"name", "route_id", "server_addr"})
+    end
+
     local entries = self.entry_buffer.entries
     table.insert(entries, entry)
+    -- add batch metric for every route
+    if batch_metrics  then
+        batch_metrics:set(#entries, prometheus.gen_arr(self.name, self.route_id, self.server_addr))
```

Comment:
We can store the table: `self.xxx = {self.name, self.route_id, self.server_addr}`.

Reply:
fixed.